sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/externalrules/CxxExternalRulesSensor.java

```diff
@@ -59,11 +72,30 @@ protected String reportPathKey() {
   public void describe(SensorDescriptor descriptor) {
     descriptor.onlyOnLanguage(CxxLanguage.KEY).name("CxxExternalRulesSensor");
   }
-  
+
+  /**
+   * {@inheritDoc}
+   */
   @Override
-  public void processReport(final SensorContext context, File report) throws XMLStreamException {
+  public void execute(SensorContext context) {
+    try {
+      transformFiles();
+    } catch (Exception e) {
```

Comment:
Think try/catch is at the wrong place. If one transformation is failing no other is executed?
Would prefer: if one transformation is failing, should try the next one